pkg/controller/remotemachineset/openstackactuator.go

```diff
@@ -134,3 +156,46 @@ func decodeOpenStackMachineProviderSpec(rawExt *runtime.RawExtension, scheme *ru
 	}
 	return spec, nil
 }
+
+// yamlOptsBuilder lets us provide our own functions to return a 'clouds.yaml' file that has been
+// unmarshaled into the format expected by the OpenStack clients.
+type yamlOptsBuilder struct {
+	cloudYaml map[string]clientconfig.Cloud
+}
+
+func newYamlOptsBuilder(kubeClient client.Client, credsSecretKey types.NamespacedName) (*yamlOptsBuilder, error) {
+
+	credsSecret := &corev1.Secret{}
+	if err := kubeClient.Get(context.TODO(), credsSecretKey, credsSecret); err != nil {
+		return nil, err
+	}
+
+	cloudsYaml, ok := credsSecret.Data[constants.OpenStackCredentialsName]
+	if !ok {
+		return nil, errors.New("did not find credentials in the ClusterDeployment's secret")
+	}
+
+	var clouds clientconfig.Clouds
+	if err := yaml.Unmarshal(cloudsYaml, &clouds); err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal yaml stored in secret")
+	}
+
+	optsBuilder := &yamlOptsBuilder{
+		cloudYaml: clouds.Clouds,
+	}
+	return optsBuilder, nil
+}
+
+func (opts *yamlOptsBuilder) LoadCloudsYAML() (map[string]clientconfig.Cloud, error) {
+	return opts.cloudYaml, nil
+}
+
+func (opts *yamlOptsBuilder) LoadSecureCloudsYAML() (map[string]clientconfig.Cloud, error) {
+	// secure.yaml is optional so just pretend it doesn't exist
+	var secureClouds clientconfig.Clouds
+	return secureClouds.Clouds, nil
```

Comment:
Isn't this just an nil map?
```suggestion
	return nil, nil
```